pyscript.core/src/plugins/py-terminal.js

```diff
@@ -0,0 +1,114 @@
+// PyScript py-terminal plugin
+import { hooks } from "../core.js";
+
+const XTERM = "5.3.0";
+const XTERM_READLINE = "1.1.1";
+
+const { assign } = Object;
+
+// Avoid conflicts with py-terminal re-definition
+if (!customElements.get("py-terminal")) {
+    document.head.append(
+        assign(document.createElement("link"), {
+            rel: "stylesheet",
+            href: `https://cdn.jsdelivr.net/npm/xterm@${XTERM}/css/xterm.min.css`,
+        }),
+    );
+
+    const [{ Terminal }, { Readline }] = await Promise.all([
+        import(
+            /* webpackIgnore: true */ `https://cdn.jsdelivr.net/npm/xterm@${XTERM}/+esm`
+        ),
+        import(
+            /* webpackIgnore: true */ `https://cdn.jsdelivr.net/npm/xterm-readline@${XTERM_READLINE}/+esm`
+        ),
+    ]);
+
+    customElements.define(
+        "py-terminal",
+        class extends HTMLElement {
+            #terminal;
+            #readline;
+            init(options) {
+                this.#readline = new Readline();
+                this.#terminal = new Terminal({
+                    theme: {
+                        background: "#191A19",
+                        foreground: "#F5F2E7",
+                    },
+                    ...options,
+                });
+                this.#terminal.loadAddon(this.#readline);
+                this.#terminal.open(this);
+                this.#terminal.focus();
+            }
+            readline(prompt) {
+                return this.#readline.read(prompt);
+            }
+            write(line) {
+                this.#readline.write(line);
+            }
+        },
+    );
+
+    const codeBefore = `
+        from pyscript import pyterminal as _pyterminal
+        _pyterminal.init()
+    `;
+
+    const codeAfter = `
+        from pyscript import pyterminal as _pyterminal
+
+        # avoid bootstrapping interact() if no terminal exists
+        if _pyterminal.PY_TERMINAL:
+            import code as _code
+            _code.interact()
```

Comment:
It's a shame not to support top level await...

Reply:
@hoodmane not intentional ... and I am not sure what you are referring to. Mind to expand a bit? We do support top level await if the tag has an `async` attribute though, but I am trying to figure out what would be the benefit here ... or maybe you are referring to some missing field when `interact(...)` is called that would allow top level await within the repl?

Thanks in advance for any sort of clarification, please remember I know nothing about Python :sweat_smile:

Reply:
OK, this has been discussed during yesterday community call, I am waiting for @hoodmane link/hint on how to workaround and enable the top level await if possible, thanks!